File: Diagrammatize.Design.Pattern/code/python/Decorator/A2/SideBorder.py
from Border import Border

class SideBorder(Border):
    def __init__(self, display, ch):
        super().__init__(display)
        self.borderChar = ch

    def getColumns(self):
        return 1 + self.display.getColumns() + 1

    def getRows(self):
        return self.display.getRows()

    def getRowText(self, row):
        return self.borderChar + self.display.getRowText(row) + self.borderChar

File: Diagrammatize.Design.Patterns/code/cxx/Decorator/Sample/SideBorder.cpp
#include "SideBorder.hpp"

SideBorder::SideBorder(Display* display, char ch):Border(display), borderChar(ch) {
}

int SideBorder::getColumns() {
    return 1 + display->getColumns() + 1;
}

int SideBorder::getRows() {
    return display->getRows();
}

std::string SideBorder::getRowText(int row) {
    return borderChar + display->getRowText(row) + borderChar;
}
